content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+We can actually go ahead and run the application by doing `npm run start`, which kicks off the build process and hosts the app on `https://localhost:1234/`. You'll see a cute animation of a character that looks like Bomberman.
+
+Let's review the code a little bit. The two TypeScript files that are important are the following: `main.ts`, on the root directory, and `src\app.ts`
+
+`main.ts` is a simple file that imports from `app.ts` and creates a GameApp object by passing an HTML element, a number for screen width, and one for screen height. I want to make a low-res kind of game so I'll drop down the resolution to _300x75_:
+
+```ts
+import { GameApp } from "./app/app"
+const myGame = new GameApp(document.body, 300, 75)
+```
+
+Finally, we have `/assets/loader.js`, which is where the creator of the template left some code that loads the game assets. Let's modify it to suit our needs:
+
+```ts
+import ghost from "./images/ghost/*.png"
+import cloud from "./images/cloud/*.png"
+import obstacle1 from "./images/obstacle1/*.png"
+import obstacle2 from "./images/obstacle2/*.png"
+import * as PIXI from "pixi.js"
+
+const spriteNames = {
+  ghost: Object.values(ghost),
+  obstacleGrave: Object.values(obstacle1),
+  obstaclePumpkin: Object.values(obstacle2),
+  cloud: Object.values(cloud),
+}
+
+export function GetSprite(name) {
+  return new PIXI.AnimatedSprite(
+    spriteNames[name].map(path => PIXI.Texture.from(path))
+  )
+}
+```
+
+We are now able to call `GetSprite()` from `app.ts` with one of the four values (specifically, the keys of `spriteNames`). The code essentially gets the files from each of the subdirectories and creates a texture from them. It's mostly an implementation detail and you can forget about it as long as you recall that the loading happens on that file.
+
+We could do something more strict/expandable here but for now this will work well for us, and is a low amount of code to maintain.
+
+**Note**: The code above assumes you have your assets in the specified folders! I'd suggest copying them from the repository.
+
+## Game Code
+
+Once the assets are out of the way, we can focus our efforts on the game code. Recall that this resides in `src\app.ts`.
+
+```ts
+export class GameApp {
+  private app: PIXI.Application
+  constructor(parent: HTMLElement, width: number, height: number) {
+    this.app = new PIXI.Application({
+      width,
+      height,
+      backgroundColor: 0x000000,
+    })
+
+    // init Pixi loader
+    let loader = new PIXI.Loader()
+
+    // Add user player assets
+    console.log("Player to load", playerFrames)
+    Object.keys(playerFrames).forEach(key => {
+      loader.add(playerFrames[key])
+    })
+
+    // Load assets
+    loader.load(this.onAssetsLoaded.bind(this))
+  }
+}
+```
```

Comment:
is this block really necessary given the fact that there is a second block below that more or less does the same thing?

Reply:
I thought about this when editing the post, but I decided to keep it since it's a good segway into mentioning some details regarding Pixi's loader.